pkg/claims/result.go

```diff
@@ -66,3 +69,21 @@ func (r Result) NewOutput(name string, data []byte) Output {
 		Value:         data,
 	}
 }
+
+func (r Result) FilterSensitiveOutputs(result driver.OperationResult, bun bundle.Bundle, store secrets.Store) []Output {
+	o := make([]Output, 0, len(result.Outputs))
+	for name, value := range result.Outputs {
+		sensitive, err := bun.IsOutputSensitive(name)
+		ot := r.NewOutput(name, []byte(value))
+		if err != nil || !sensitive {
+			o = append(o, ot)
+			continue
+		}
+
+		ot = ot.FormatSensitive()
+		o = append(o, ot)
+		store.Create(secrets.SourceSecret, string(ot.Value), string(value))
```

Comment:
The error isn't being handled here.

Since this function is persisting sensitive outputs, the name of the function should be changed to make it clear that it is going to do that. Otherwise it sounds like it's just filtering and is safe to call anytime.